hudi-common/src/main/java/org/apache/hudi/common/model/BaseAvroPayload.java

```diff
@@ -68,7 +73,21 @@ public Comparable getOrderingVal() {
    * We will not do deserialization in this method.
    */
   public boolean isDeleted(Schema schema, Properties props) {
-    return isDeletedRecord;
+    return isDeletedRecord
+        || HoodieRecordUtils.isCustomDeleteRecord(
+            recordBytes,
+            schema,
+            props,
+            recordData -> {
+              try {
+                GenericRecord data = HoodieAvroUtils.bytesToAvro(recordData, schema);
```

Comment:
We should not deserialize the record bytes here, which causes significant overhead per record.  Only determine `isDeletedRecord` inside the constructor.